medusa/session/factory.py

```diff
@@ -0,0 +1,26 @@
+"""Session class factory methods."""
+
+import logging
+from cachecontrol import CacheControlAdapter
+from cachecontrol.cache import DictCache
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def add_cache_control(session, cache_control_config):
+    if not all(['cache_etags' in cache_control_config,
+                'serializer' in cache_control_config,
+                'heuristic' in cache_control_config]):
+        logging.warning('Insufficient paramaters provided for the add_cache_control factory.')
+        return
```

Comment:
Instead of a warning and a return None this should be an assert since we want to assert that the code is implemented properly.  Also this could be simplified using set syntax.
```python
def add_cache_control(session, cache_control_config):
    assert {'cache_etags', 'serializer', 'heuristic'}.issubset(cache_control_config.keys), cache_control_config
    ...
```